Sources/TuistXCActivityLog/XCActivityLogController.swift

```diff
@@ -167,11 +169,71 @@ public struct XCActivityLogController: XCActivityLogControlling {
                 errorCount: buildStep.errorCount
             ),
             category: category,
-            issues: issues
+            issues: issues,
+            files: files,
+            targets: targets.map {
+                let targetSteps = flattenedXCLogParserBuildStep([$0.step])
+                return XCActivityTarget(
+                    name: $0.name,
+                    project: targetSteps.first(where: { $0.project() != "" })?.project() ?? "",
+                    buildDuration: Int(($0.step.duration * 1000).rounded(.up)),
+                    compilationDuration: Int(($0.step.compilationDuration * 1000).rounded(.up)),
+                    status: targetSteps.allSatisfy { $0.errorCount == 0 } ? .success : .failure
+                )
+            }
         )
     }
 
-    private func issues(steps: [BuildStep], targets _: [XCActivityTarget]) async throws
+    private func files(steps: [BuildStep]) async throws
+        -> [XCActivityBuildFile]
+    {
+        var seenPaths: Set<RelativePath> = Set()
+        return
+            try await steps
+                .filter { $0.type == .detail }
+                .concurrentCompactMap { step -> XCActivityBuildFile? in
+                    let type: XCActivityBuildFileType
+                    switch step.detailStepType {
+                    case .swiftCompilation:
+                        type = .swift
+                    case .cCompilation:
+                        type = .c
+                    default:
+                        if step.signature.hasPrefix("SwiftCompile ") {
+                            type = .swift
+                        } else {
+                            return nil
+                        }
+                    }
+
+                    if let absolutePath = try? AbsolutePath(
+                        validating: step.documentURL
+                            .replacingOccurrences(of: "file://", with: "")
+                    ), DeveloperEnvironment.current.derivedDataDirectory.isAncestor(of: absolutePath) {
+                        return nil
+                    }
+
+                    guard let path = try await path(of: step) else { return nil }
+
+                    return XCActivityBuildFile(
+                        type: type,
+                        target: step.target(),
+                        project: step.project(),
+                        path: path,
+                        compilationDuration: Int((step.compilationDuration * 1000).rounded(.up))
+                    )
+                }
+                .filter { buildFile in
+                    if seenPaths.contains(buildFile.path) {
+                        return false
+                    } else {
+                        seenPaths.insert(buildFile.path)
+                        return true
+                    }
+                }
```

Comment:
Why are we doing this filter after concurrently compact-mapping? It seems that this logic here can potentially yield different results depending on the order of the files returned by the compact map.

Reply:
There is an "Emit Swift module" and "Compile file..." steps for the same file that are both categorized as Swift compilation. I updated the compact map to filter out those `Emit Swift module` steps as we don't care about those in this context and removed this filter.